lib/qesapdeployment.pm

```diff
@@ -545,4 +585,102 @@ sub qesap_remote_hana_public_ips {
     return @{$data->{hana_public_ip}->{value}};
 }
 
+=head3 qesap_wait_for_ssh
+
+  Probe specified port on the remote host each 5sec till response.
+  Return -1 in case of timeout
+  Return total time of retry loop in case of pass.
+
+=over 3
+
+=item B<HOST> - IP of the host to probe
+
+=item B<TIMEOUT> - time to wait before to give up, default is 10mins
+
+=item B<PORT> - port to probe, default is 22
+
+=back
+=cut
+
+sub qesap_wait_for_ssh {
+    my (%args) = @_;
+    croak 'Missing mandatory host argument' unless $args{host};
+    $args{timeout} ||= bmwqemu::scale_timeout(600);
+    $args{port} ||= 22;
+    my $start_time = time();
+    my $check_port = 1;
+
+    # Looping until reaching timeout or passing two conditions :
+    # - SSH port 22 is reachable
+    # - journalctl got message about reaching one of certain targets
+    while ((my $duration = time() - $start_time) < $args{timeout}) {
+        if ($check_port) {
+            $check_port = 0 if (script_run(join(' ', 'nc', '-vz', '-w', '1', $args{host}, $args{port}), quiet => 1) == 0);
+        }
+        else {
+            return $duration;
+        }
+        sleep 5;
+    }
+    return -1;
+}
+
+=head3 qesap_cluster_log_cmds
+
+  List of commands to collect logs from a deployed cluster
+
+=cut
+
+sub qesap_cluster_log_cmds {
+    return (
+        {
+            Cmd => 'crm status',
+            Logs => ['crm_status.txt'],
+        },
+        {
+            Cmd => 'crm configure show',
+            Logs => ['crm_configure.txt'],
+        },
+        {
+            Cmd => 'lsblk -i -a',
+            Logs => ['lsblk.txt'],
+        },
+        {
+            Cmd => 'journalctl -b --no-pager -o short-precise',
+            Logs => ['journalctl.txt'],
+        },
+        {
+            Cmd => 'systemctl --no-pager --full status sbd',
+            Logs => ['sbd.txt'],
+        },
+    );
+}
+
+=head3 qesap_cluster_logs
+
+  Collect logs from a deployed cluster
+
+=cut
+
+sub qesap_cluster_logs {
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my $inventory = qesap_get_inventory($prov);
+    if (script_run("test -e $inventory") == 0)
+    {
+        # sleep bmwqemu::scale_timeout(300)
+        foreach my $host ('vmhana01', 'vmhana02') {
+            foreach my $cmd (qesap_cluster_log_cmds()) {
```

Comment:
Is there an advantage with having this `qesap_cluster_log_cmds` which returns a HASH vs. defining the HASH as a variable inside `qesap_cluster_logs`?

Reply:
yes, It is the first step to unify the code with `ha_export_logs` that I plan to split in two functions too